git/git-clone.yaml

```diff
@@ -39,14 +43,35 @@ spec:
     image: gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:latest
     script: |
       CHECKOUT_DIR="$(workspaces.output.path)/$(inputs.params.subdirectory)"
+
+      cleandir() {
+        # Delete any existing contents of the repo directory if it exists.
+        #
+        # We don't just "rm -rf $CHECKOUT_DIR" because $CHECKOUT_DIR might be "/"
+        # or the root of a mounted volume.
+        if [[ -d "$CHECKOUT_DIR" ]] ; then
+          # Delete non-hidden files and directories
+          rm -rf "$CHECKOUT_DIR"/*
+          # Delete files and directories starting with . but excluding ..
+          rm -rf "$CHECKOUT_DIR"/.[!.]*
```

Comment:
what about just simply : `rm -rf ${CHECKOUT_DIR} && mkdir -p ${CHECKOUT_DIR}` ?

Reply:
I am not sure what would happen if CHECKOUT_DIR is `/` or if it is a mount point of a volume.  I figured it was safer to wipe out the contents rather than the directory itself.  But I'm still a relative linux noob so if you think it's ok to do that then sounds good to me.